app/views/admin/products_v3/_product_row.html.haml

```diff
@@ -1,3 +1,5 @@
+%template{ id: "producer_options"}
+  = options_for_select(producer_options, product.supplier_id)
```

Comment:
I just came across this PR and had a quick look. Thanks for working on this!

This partial is repeated for every product row, yet it appears the intention was to only include the template once. When the controller searches for the element by ID, it will simply find the last one (standard browser behaviour).

This is ok, it would work, and as Maikel mentioned the compression will take care of most of it. But it looks like it should be very easy to remove the duplication by moving the template to another partial. So if you have time, feel free to go ahead update it ;)

Reply:
Aaah, my bad. I just pushed a correction.

Reply:
Thanks, but i can see:
* this is still repeated for every product.
* `producer_options` is no longer needed in `product_row`
* `options_for_select` doesn't need a second parameter for selected value (tomselect will need to choose it for each product when initialised)

But first let's see if we can confirm a performance improvement before progressing.